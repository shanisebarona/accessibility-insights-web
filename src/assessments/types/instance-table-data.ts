// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { GeneratedAssessmentInstance } from 'common/types/store-data/assessment-result-data';

export type InstanceTableHeaderType = 'none' | 'default';

export interface InstanceTableRow<P = {}> {
    key: string;
    statusChoiceGroup: JSX.Element;
    visualizationButton?: JSX.Element;
    instance: GeneratedAssessmentInstance<P>;
}

export interface InstanceTableColumn {
    key: string;
    name: string;
    onRender: (item: InstanceTableRow) => JSX.Element;
}
