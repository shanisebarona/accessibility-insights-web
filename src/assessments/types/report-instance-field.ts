// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    ColumnValue,
    ColumnValueBag,
    isScalarColumnValue,
} from 'common/types/property-bag/column-value-bag';
import { PropertyBagColumnRendererConfig } from 'common/types/property-bag/property-bag-column-renderer-config';
import { TestStepInstance } from 'common/types/store-data/assessment-result-data';

export type ReportInstanceField = {
    key: string;
    label: string;
    getValue: (instance: Partial<TestStepInstance>) => ColumnValue;
};
export type ReportInstanceFieldMap = { [KEY in string]?: ReportInstanceField };

export type ReportInstanceFields = ReportInstanceField[];

type HasPropertyBag<PB> = { propertyBag: PB };
type PropertyBagKey<PB> = PB[keyof PB];

function fromPropertyBagField<PB>(label: string, key: keyof PB & string): ReportInstanceField {
    function getValue(i: HasPropertyBag<PB>): string {
        return i.propertyBag && i.propertyBag[key] && (i.propertyBag[key] as any).toString();
    }
    return { key, label, getValue };
}

function fromColumnValueBagField<PB extends ColumnValueBag>(
    label: string,
    key: keyof PB & string,
    defaultValue = '-',
): ReportInstanceField {
    function getValue(i: HasPropertyBag<PB>): string | PropertyBagKey<PB> {
        const value = i.propertyBag && i.propertyBag[key];
        return isValid(value) ? value : defaultValue;
    }

    return { key, label, getValue };
}

function fromPropertyBagFunction<PB>(
    label: string,
    key: string,
    accessor: (bag: PB) => string,
): ReportInstanceField {
    function getValue(i: HasPropertyBag<PB>): string {
        return i.propertyBag && accessor(i.propertyBag);
    }
    return { key, label, getValue };
}

const common: ReportInstanceFieldMap = {
    comment: { key: 'comment', label: 'Comment', getValue: i => i.description },
    snippet: { key: 'snippet', label: 'Snippet', getValue: i => i.html },
    path: { key: 'path', label: 'Path', getValue: i => i.target && i.target.join(', ') },
    manualSnippet: { key: 'manualSnippet', label: 'Code Snippet', getValue: i => i.html },
    manualPath: { key: 'manualPath', label: 'CSS Selector', getValue: i => i.selector },
};

function isValid(value: ColumnValue): ColumnValue {
    if (!value) {
        return false;
    }
    if (!isScalarColumnValue(value) && Object.keys(value).length === 0) {
        return false;
    }
    return true;
}

function fromColumns<T extends ColumnValueBag>(
    cfg: PropertyBagColumnRendererConfig<T>[],
): ReportInstanceField[] {
    return cfg.map(fromColumnConfig);

    function fromColumnConfig({
        propertyName,
        defaultValue,
        displayName,
    }: PropertyBagColumnRendererConfig<T>): ReportInstanceField {
        const getValue = (inst: HasPropertyBag<T>) => {
            const value = inst.propertyBag && inst.propertyBag[propertyName];
            return isValid(value) ? value : defaultValue;
        };

        return {
            key: propertyName,
            label: displayName,
            getValue,
        };
    }
}

function fromSnippet(key: string, label: string): ReportInstanceField {
    return { key, label, getValue: i => i.html };
}

export const ReportInstanceField = {
    fromPropertyBagField,
    fromColumnValueBagField,
    fromPropertyBagFunction,
    fromColumns,
    fromSnippet,
    common,
};
