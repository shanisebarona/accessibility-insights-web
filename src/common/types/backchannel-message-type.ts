// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

export type BackchannelStoreRequestMessage = {
    messageId: string; // same as WindowMessage
    messageType: 'backchannel_window_message.store_request';
    stringifiedMessageData: string; // JSON.stringify(originalMessage)
};

export type BackchannelRetrieveRequestMessage = {
    messageId: string; // same as WindowMessage
    messageType: 'backchannel_window_message.retrieve_request';
};

export type BackchannelRetrieveResponseMessage = {
    messageId: string; // same as WindowMessage
    messageType: 'backchannel_window_message.retrieve_response';
    stringifiedMessageData: string;
};
