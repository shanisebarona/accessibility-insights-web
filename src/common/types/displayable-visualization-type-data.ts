// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
export interface DisplayableVisualizationTypeData {
    title: string;
    subtitle?: JSX.Element;
    enableMessage: string;
    toggleLabel: string;
    linkToDetailsViewText: string;
}
