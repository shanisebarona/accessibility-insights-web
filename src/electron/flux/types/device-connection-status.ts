// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
export enum DeviceConnectionStatus {
    Unknown,
    Connected,
    Disconnected,
}
