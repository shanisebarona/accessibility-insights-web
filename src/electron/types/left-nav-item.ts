// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { LeftNavItemKey } from './left-nav-item-key';

export type LeftNavItem = {
    key: LeftNavItemKey;
    displayName: string;
    featureFlag?: string;
    onSelect: () => void;
};
