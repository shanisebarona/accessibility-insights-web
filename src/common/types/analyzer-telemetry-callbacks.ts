// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AxeAnalyzerResult } from 'common/types/axe-analyzer-result';
import { VisualizationType } from 'common/types/visualization-type';
import {
    RuleAnalyzerScanTelemetryData,
    IssuesAnalyzerScanTelemetryData,
    NeedsReviewAnalyzerScanTelemetryData,
} from '../extension-telemetry-events';

export type IAnalyzerTelemetryCallback = ForRuleAnalyzerScanCallback;

export type ForRuleAnalyzerScanCallback = (
    analyzerResult: AxeAnalyzerResult,
    scanDuration: number,
    elementsScanned: number,
    testVisualizationType: VisualizationType,
    requirementName?: string,
) => RuleAnalyzerScanTelemetryData;

export type ForIssuesAnalyzerScanCallback = (
    analyzerResult: AxeAnalyzerResult,
    scanDuration: number,
    elementsScanned: number,
    testVisualizationType: VisualizationType,
) => IssuesAnalyzerScanTelemetryData;

export type ForNeedsReviewAnalyzerScanCallback = (
    analyzerResult: AxeAnalyzerResult,
    scanDuration: number,
    elementsScanned: number,
    testVisualizationType: VisualizationType,
) => NeedsReviewAnalyzerScanTelemetryData;
