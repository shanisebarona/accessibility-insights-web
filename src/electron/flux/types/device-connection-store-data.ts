// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { DeviceConnectionStatus } from 'electron/flux/types/device-connection-status';

export interface DeviceConnectionStoreData {
    status: DeviceConnectionStatus;
}
