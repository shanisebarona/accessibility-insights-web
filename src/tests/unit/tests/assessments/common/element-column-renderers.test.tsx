// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import {
    onRenderPathColumn,
    onRenderSnippetColumn,
} from 'assessments/common/element-column-renderers';
import { InstanceTableRow } from 'assessments/types/instance-table-data';

describe('element column renderers', () => {
    const testCases: any[] = [
        ['null target', null, null],
        ['one target', ['A'], 'X'],
        ['two targets', ['A', 'B'], 'XY'],
    ];

    describe(onRenderSnippetColumn, () => {
        it.each(testCases)('%s', (testName, target, html) => {
            const item = createRowData(target, html);

            expect(onRenderSnippetColumn(item)).toMatchSnapshot();
        });
    });

    describe(onRenderPathColumn, () => {
        it.each(testCases)('%s', (testName, target, html) => {
            const item = createRowData(target, html);

            expect(onRenderPathColumn(item)).toMatchSnapshot();
        });
    });
});

function createRowData(target: string[], html: string): InstanceTableRow {
    return {
        statusChoiceGroup: null,
        visualizationButton: null,
        instance: {
            target: target,
            html: html,
            testStepResults: null,
        },
        key: 'key',
    };
}
