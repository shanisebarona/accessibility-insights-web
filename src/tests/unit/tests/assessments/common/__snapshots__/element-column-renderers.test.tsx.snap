// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`element column renderers onRenderPathColumn null target 1`] = `
<AssessmentInstanceDetailsColumn
  customClassName="not-applicable"
  textContent=""
/>
`;

exports[`element column renderers onRenderPathColumn one target 1`] = `
<AssessmentInstanceDetailsColumn
  customClassName="not-applicable"
  textContent="A"
/>
`;

exports[`element column renderers onRenderPathColumn two targets 1`] = `
<AssessmentInstanceDetailsColumn
  customClassName="not-applicable"
  textContent="A;B"
/>
`;

exports[`element column renderers onRenderSnippetColumn null target 1`] = `
<AssessmentInstanceDetailsColumn
  customClassName="not-applicable"
  textContent={null}
/>
`;

exports[`element column renderers onRenderSnippetColumn one target 1`] = `
<AssessmentInstanceDetailsColumn
  customClassName="not-applicable"
  textContent="X"
/>
`;

exports[`element column renderers onRenderSnippetColumn two targets 1`] = `
<AssessmentInstanceDetailsColumn
  customClassName="not-applicable"
  textContent="XY"
/>
`;
